src/Microsoft.AspNet.Authentication.Facebook/FacebookAppBuilderExtensions.cs

```diff
@@ -44,13 +23,15 @@ public static IApplicationBuilder UseFacebookAuthentication(this IApplicationBui
             {
                 throw new ArgumentNullException(nameof(app));
             }
-
-            var options = new FacebookOptions();
-            if (configureOptions != null)
+            if (configureOptions == null)
             {
-                configureOptions(options);
+                throw new ArgumentNullException(nameof(configureOptions));
```

Comment:
Only if the default options are meaningful. That's not the case for any of the security middleware. Also, don't use opational params, add an extra overload.